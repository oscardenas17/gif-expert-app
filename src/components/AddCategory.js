import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setinputValue] = useState(''); // '  '

    const handleInputChange = (e) =>{
        // console.log(e.target.value);
        setinputValue(e.target.value);

        console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) =>{
        e.preventDefault();

        if( inputValue.trim().length > 2){

            setCategories( cate => [ inputValue, ...cate] );
            setinputValue('');
        }

    }

    return (
        <form onSubmit = {handleSubmit}>
            <p> { inputValue}</p>
           <input
                 type="text"
                 value = {inputValue}
                 onChange = {handleInputChange}

           />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
