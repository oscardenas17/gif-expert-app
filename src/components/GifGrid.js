import React from 'react'

export const GifGrid = ( {category}) => {
  
  const getGifs = async() =>{

    const url = 'https://api.giphy.com/v1/gifs/search?q=luffy&limit=10&api_key=NsfM7mcDnuC60HeX26qU7ujruT4Jk2Gg'
    const resp = await fetch(url);
    const {data} = await resp.json();

    const gifs = data.map( img =>{
        return{
            id: img.id,
            title: img.title,
            url: img.images?.downsized_medium.url
        }
    })

    console.log(gifs);


  }
  
  getGifs();

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}
