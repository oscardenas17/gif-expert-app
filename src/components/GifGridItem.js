import React from 'react'

export const GifGridItem = ( {id,title, url} ) => {
    // console.log( id, title, url);

    return (
        <div>
          <img src= {url} alt={title} className="source" />
          <p>{title}</p>
        </div>
    )
}
