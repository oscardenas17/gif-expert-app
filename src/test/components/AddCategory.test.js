import React from 'react';
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Prueba en <AddCategory />', () => {
    
    const setCategories = () => {};
        
    const wrapper =  shallow( <AddCategory setCategories={setCategories}/>);

    test('Se muestra correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    })


    test('Cambios en cada de texto', () => {
        
       const input = wrapper.find('input');
        const value = 'hola mundo'
       input.simulate('change', { target: { value:value}});

       expect(wrapper.find('p').text().trim()).toBe(value);
    })
    
})
