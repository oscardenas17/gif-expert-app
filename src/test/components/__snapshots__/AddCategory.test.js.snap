// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <AddCategory /> Se muestra correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
